### PR TITLE
Add Redpen support

### PR DESCRIPTION
**Pull request by rhysd**
I added support for [redpen](http://redpen.cc/), a proofreading tool for writers and programmers. It supports multi document formats (markdown, latex, reST and asciidoc). I usually use redpen for markdown so I added it as a markdown linter at first.

### CODE REVIEW

File: ale_linters/markdown/redpen.vim [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+function! ale_linters#markdown#redpen#HandleErrors(buffer, lines) abort
+    " Only one file was passed to redpen. So response array has only one
+    " element.
+    let l:res = json_decode(join(a:lines))[0]
+    let l:errors = []
+    for l:err in l:res.errors
+        if has_key(l:err, 'startPosition')
+            let l:lnum = l:err.startPosition.lineNum
+            let l:col = l:err.startPosition.offset
```
Comment (w0rp) [inline ale_linters/markdown/redpen.vim]: While you're at it, could you add a condition below for reading the `endPosition` too? You can set the `end_lnum` and `end_col` keys. Then we'll be able to draw better error highlights.

Comment (rhysd) [reply ale_linters/markdown/redpen.vim]: sure.

**Comment by rhysd** [comment]
@w0rp I added support for `end_lnum` and `end_col`.

**Comment by rhysd** [comment]
@w0rp I added support for `end_lnum` and `end_col`.

**Comment by rhysd** [comment]
ah, test failed

```
  Starting Vader: /testplugin/test/handler/test_redpen_handler.vader
    (1/2) [EXECUTE] redpen handler should handle errors output
    (1/2) [EXECUTE] (X) Vim(let):E474: Trailing comma: }     ]   } ]
      > function ale_linters#markdown#redpen#HandleErrors, line 3
    (2/2) [EXECUTE] redpen handler should no error output
  Success/Total: 1/2
```

It does not reproduce in my local...

**Comment by w0rp** [comment]
Okay, now NeoVim is happy. Cheers! :beers:

**Pull request merged**